### DISCUSSION TITLE
Select multiple records based on enum

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#9411] [Help]
I'm using **laravel-enum** to create enum classes I'm trying to use `->multiple()` to select more than one option but it did not work

**I get "Array to string conversion"**

```
Select::make('status')
->label(__('Status'))
->options(UserStatus::getKeys())
->default(UserStatus::getKey(UserStatus::Active))
->required()
->native(false)
->searchable()
->multiple(),
```

**User.php**

```
protected $casts = [
    'status' => UserStatus::class,
];
```

**Migration File**

```
$table->enum('status', UserStatus::getValues())
                ->default(UserStatus::Active);
```

**Comment by ManojKiranA** [comment]
if you status column is array of enum values then you need to take a look [AsEnumCollection](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Eloquent/Casts/AsEnumCollection.php#L10)

and [Casting Arrays Of Enums](https://laravel.com/docs/10.x/eloquent-mutators#casting-arrays-of-enums) docs

```
use Illuminate\Database\Eloquent\Casts\AsEnumCollection;

protected $casts = [
    'status' =>  AsEnumCollection::class.':'.UserStatus::class,
];

```

![image](https://github.com/filamentphp/filament/assets/30294553/daabf901-8d94-4881-98ef-aa63dae2d9a5)
